### PR TITLE
Tighten DS rollout catching function

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
#### What type of PR is this?
/kind flake

#### Which issue(s) this PR fixes:

Fixes # partially https://github.com/kubernetes/kubernetes/issues/100138 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ravisantoshgudimetla** [comment]
/sig apps

@soltysh

**Comment by ravisantoshgudimetla** [comment]
By the time we're checking for [pod hashes](https://github.com/kubernetes/kubernetes/pull/100345/files#diff-f244160bd6f99fe973d578f2af102fc06070e1fdf879b9bb2f017de6fd9a070aR1085) the previous version of DS might still be running. So ignoring the pods that are going to be deleted should be ok.

File: test/e2e/apps/daemon_set.go [hunk 1]
```diff
@@ -1070,7 +1074,7 @@ func checkDaemonPodsImageAndAvailability(c clientset.Interface, ds *appsv1.Daemo
 		// Make sure every daemon pod on the node has been updated
 		nodeNames := schedulableNodes(c, ds)
 		for _, node := range nodeNames {
-			if nodesToUpdatedPodCount[node] == 0 {
+			if nodesToUpdatedPodCount[node] != 1 {
```
Comment (soltysh) [inline test/e2e/apps/daemon_set.go:1077]: Don't change that bit, I'd only leave the `DeletionTimestamp` checks you've added in those 2 places. This wasn't previously the issue, this was introduced with MaxSurge for rolling update in https://github.com/kubernetes/kubernetes/pull/96441

**Review by soltysh — CHANGES_REQUESTED** [review]

File: test/e2e/apps/daemon_set.go (hunk 1 above)
Comment (ravisantoshgudimetla) [reply test/e2e/apps/daemon_set.go]: I see, even though this doesn't hurt. I have removed it.

**Review by soltysh — APPROVED** [review]
/priority important-soon
/triage accepted
/lgtm
/approve
/milestone v1.21
since this is fixing a flaking test

**Pull request merged**